src/components/radio/__tests__/Radio.spec.js

```diff
@@ -0,0 +1,127 @@
+import { clearContext, render } from "@/components/_util/testHelpers";
+import Radio from "../Radio.svelte";
+import { writable } from "svelte/store";
+
+describe("Radio component", () => {
+  afterEach(() => {
+    clearContext();
+  });
+  test("should render default", () => {
+    const { container } = render(Radio);
+    expect(container.innerHTML).toContain("ant-radio");
+  });
+
+  test("should render with default state", () => {
+    const { container } = render(Radio, { defaultChecked: true });
+
+    const input = container.getElementsByTagName("INPUT")[0];
+    expect(input.checked).toEqual(true);
+  });
+
+  test("should render with checked defined", () => {
+    const { container } = render(Radio, { checked: true });
+
+    const input = container.getElementsByTagName("INPUT")[0];
+    expect(input.checked).toEqual(true);
+  });
+
+  test("should render with disabled state", () => {
+    const { container } = render(Radio, { disabled: true });
+    const input = container.getElementsByTagName("INPUT")[0];
+    expect(input.disabled).toEqual(true);
+  });
+
+  test("should trigger change event if change happens", () => {
+    const { container, component } = render(Radio);
+    const onChange = jest.fn();
+    component.$on("change", onChange);
+    const input = container.getElementsByTagName("INPUT")[0];
+    const evt = document.createEvent("HTMLEvents");
+    evt.initEvent("change", false, true);
+    input.checked = true;
+    input.dispatchEvent(evt);
+    expect(onChange).toHaveBeenCalled();
+  });
+
+  test("should trigger mouseenter when over the label", () => {
+    const { container, component } = render(Radio);
+    const onMouseOver = jest.fn();
+    component.$on("mouseenter", onMouseOver);
+    const label = container.getElementsByTagName("LABEL")[0];
+    const evt = document.createEvent("HTMLEvents");
+    evt.initEvent("mouseenter", false, true);
+    label.dispatchEvent(evt);
```

Comment:
You could also use `fireEvent` here https://testing-library.com/docs/dom-testing-library/api-events

This is fine though. I've just been using that in my tests.

Reply:
I'll start using this from now on.